source/val/validate_ray_tracing_reorder.cpp

```diff
@@ -62,6 +63,31 @@ spv_result_t ValidateHitObjectPointer(ValidationState_t& _,
   return SPV_SUCCESS;
 }
 
+spv_result_t ValidateHitObjectPointerEXT(ValidationState_t& _,
+                                         const Instruction* inst,
+                                         uint32_t hit_object_index) {
+  const uint32_t hit_object_id = inst->GetOperandAs<uint32_t>(hit_object_index);
+  auto variable = _.FindDef(hit_object_id);
+  const auto var_opcode = variable->opcode();
+  if (!variable || (var_opcode != spv::Op::OpVariable &&
+                    var_opcode != spv::Op::OpFunctionParameter &&
+                    var_opcode != spv::Op::OpAccessChain)) {
+    return _.diag(SPV_ERROR_INVALID_DATA, inst)
+           << "Hit Object must be a memory object declaration";
+  }
```

Comment:
A memory object declaration would be a variable or function parameter. So access chain doesn't really match the message.

Separately though, the spec says:
> 'Hit Object' must be a pointer to hit object used to reorder threads.

So I don't see where this restriction comes from.